app/mixins/event-wizard.js

```diff
@@ -48,28 +48,33 @@ export default Mixin.create(MutableArray, CustomFormMixin, {
   async saveEventData(propsToSave = []) {
     const event = this.get('model.event');
     const data = {};
-    for (const property of propsToSave) {
+    const results = await Promise.allSettled(propsToSave.map(property => {
       try {
-        data[property] = await event.get(property);
+        return event.get(property);
       } catch (e) {
         if (!(e.errors && e.errors.length && e.errors.length > 0 && e.errors[0].status === 404)) {
           // Lets just ignore any 404s that might occur. And throw the rest for the caller fn to catch
           throw e;
         }
       }
-    }
+    }));
+    results.then(promises => {
+      for (const [index, property] of propsToSave.entries()) {
```

Comment:
No need to extract index

Reply:
Then how can I assign the value of ```event.get(property)``` without using the ```index```?

Reply:
By using property